internal/config/main.go

```diff
@@ -46,13 +86,62 @@ func (c *Config) FlagSet() *pflag.FlagSet {
 	fs := pflag.NewFlagSet("opentelemetry-operator", pflag.ExitOnError)
 	pflag.StringVar(&c.collectorImage,
 		"otelcol-image",
-		fmt.Sprintf("quay.io/opentelemetry/opentelemetry-collector:v%s", c.v.OpenTelemetryCollector),
+		c.collectorImage,
 		"The default image to use for OpenTelemetry Collector when not specified in the individual custom resource (CR)",
 	)
 
 	return fs
 }
 
+// StartAutoDetect attempts to automatically detect relevant information for this operator. This will block until the first
+// run is executed and will schedule periodic updates.
+func (c *Config) StartAutoDetect() error {
+	err := c.AutoDetect()
+	go c.periodicAutoDetect()
+
+	return err
+}
+
+func (c *Config) periodicAutoDetect() {
+	ticker := time.NewTicker(c.autoDetectFrequency)
+
+	for range ticker.C {
+		c.AutoDetect()
+	}
+}
+
+// AutoDetect attempts to automatically detect relevant information for this operator.
+func (c *Config) AutoDetect() error {
+	changed := false
```

Comment:
As this is a no-op once the platform has been set, do we want to stop the ticker? Only needs to continue once other auto-detection capabilities are required. Or do you prefer to just leave running as minimal overhead.

Reply:
Good question. I think I'll keep this, unless we find a problem. I think one of the next features will be to detect CRDs in the cluster (Jaeger comes to my mind), and CRDs might not exist during the initial lookup, but might be available later on.